pkg/datapath/types/loader.go

```diff
@@ -83,3 +74,10 @@ type IptablesManager interface {
 	// See comments for InstallNoTrackRules.
 	RemoveNoTrackRules(IP string, port uint16, ipv6 bool) error
 }
+
+type CompilationLock interface {
```

Comment:
Can you document why this is a separate type please? I imagine it has something to do with the fact that it's taken from all over the place. Future me would also very much be interested to know at which point we could get rid of this.